service/service.go

```diff
@@ -363,3 +385,43 @@ func pdataFromSdk(res *sdkresource.Resource) pcommon.Resource {
 	}
 	return pcommonRes
 }
+
+func disableHighCardinalityMetrics() []config.View {
+	var views []config.View
+	if disableHighCardinalityMetricsFeatureGate.IsEnabled() {
+		return views
+	}
+	return []config.View{
+		{
+			Selector: &config.ViewSelector{
+				MeterName: ptr("go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"),
+			},
+			Stream: &config.ViewStream{
+				AttributeKeys: &config.IncludeExclude{
+					Excluded: []string{
+						semconv118.AttributeNetSockPeerAddr,
+						semconv118.AttributeNetSockPeerPort,
+						semconv118.AttributeNetSockPeerName,
+					},
+				},
+			},
+		},
+		{
+			Selector: &config.ViewSelector{
+				MeterName: ptr("go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"),
+			},
+			Stream: &config.ViewStream{
+				AttributeKeys: &config.IncludeExclude{
+					Excluded: []string{
+						semconv118.AttributeNetHostName,
+						semconv118.AttributeNetHostPort,
+					},
+				},
+			},
+		},
+	}
+}
+
+func ptr[T any](v T) *T {
+	return &v
```

Comment:
why can't go just provide this natively :-(

Reply:
is this a question or a comment about go?

Reply:
yes